kola/harness.go

```diff
@@ -371,6 +371,21 @@ func RunTests(patterns []string, channel, offering, pltfrm, outputDir string, ss
 			plog.Fatal(err)
 		}
 
+		if version.LessThan(semver.Version{Major: 3034}) && pltfrm == "azure" {
+			AzureOptions.UseCustomData = true
+
+			// We need to recreate the flight to apply this modification.
+			flight, err = NewFlight(pltfrm)
```

Comment:
Creating a new flight is not so good, it creates a new resource group and would also upload the image again. Since we know that `flight` is an Azure flight, can we access `flight.api.GetOpts()` and set it there? Maybe we need to add a getter to the flight or do some reflection?

Reply:
So I've been for reflection - it's far from being perfect and I had to export the `Api` and the `Opts` fields (otherwise the field that we want to update can't be set).
`api` can't be accessed as it's not part of the initial `platform.Flight` struct (and some platforms do not have this field neither). The whole thing would need a bit of abstraction.